packages/eas-cli/src/commands/share.ts

```diff
@@ -0,0 +1,305 @@
+import path from 'path';
+import tar from 'tar';
+
+import { getBuildLogsUrl } from '../build/utils/url';
+import EasCommand from '../commandUtils/EasCommand';
+import { ExpoGraphqlClient } from '../commandUtils/context/contextUtils/createGraphqlClient';
+import { EASNonInteractiveFlag } from '../commandUtils/flags';
+import {
+  DistributionType,
+  LocalBuildArchiveSourceType,
+  UploadSessionType,
+} from '../graphql/generated';
+import { FingerprintMutation } from '../graphql/mutations/FingerprintMutation';
+import { LocalBuildMutation } from '../graphql/mutations/LocalBuildMutation';
+import { toAppPlatform } from '../graphql/types/AppPlatform';
+import Log from '../log';
+import { promptAsync } from '../prompts';
+import { uploadFileAtPathToGCSAsync } from '../uploads';
+import { createProgressTracker } from '../utils/progress';
+
+export default class BuildUpload extends EasCommand {
+  static override description = 'upload a local build and generate a sharable link';
+
+  static override flags = {
+    platform: Flags.enum<Platform.IOS | Platform.ANDROID>({
+      char: 'p',
+      options: [Platform.IOS, Platform.ANDROID],
+    }),
+    'build-path': Flags.string({
+      description: 'Path for the local build',
+    }),
+    fingerprint: Flags.string({
+      description: 'Fingerprint hash of the local build',
+    }),
+    ...EASNonInteractiveFlag,
+  };
+
+  static override contextDefinition = {
+    ...this.ContextOptions.ProjectId,
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  async runAsync(): Promise<void> {
+    const { flags } = await this.parse(BuildUpload);
+    const { 'build-path': buildPath, fingerprint: manualFingerprintHash } = flags;
+    const {
+      projectId,
+      loggedIn: { graphqlClient },
+    } = await this.getContextAsync(BuildUpload, {
+      nonInteractive: false,
+    });
+
+    const platform = await this.selectPlatformAsync(flags.platform);
+    const localBuildPath = await resolveLocalBuildPathAsync(platform, buildPath);
+
+    const {
+      fingerprintHash: buildFingerprintHash,
+      developmentClient,
+      simulator,
+    } = await extractAppMetadataAsync(localBuildPath, platform);
+
+    let fingerprint = manualFingerprintHash ?? buildFingerprintHash;
+    if (fingerprint) {
+      if (
+        manualFingerprintHash &&
+        buildFingerprintHash &&
+        manualFingerprintHash !== buildFingerprintHash
+      ) {
+        const selectedAnswer = await promptAsync({
+          name: 'fingerprint',
+          message: `The provided fingerprint hash ${manualFingerprintHash} does not match the fingerprint hash of the build ${buildFingerprintHash}. Which fingerprint do you want to use?`,
+          type: 'select',
+          choices: [
+            { title: manualFingerprintHash, value: manualFingerprintHash },
+            { title: buildFingerprintHash, value: buildFingerprintHash },
+          ],
+        });
+        fingerprint = String(selectedAnswer.fingerprint);
```

Comment:
why is it needed?

Reply:
promptAsync answers are typed as `any`, so we need to do this to prevent TS from complaining
https://github.com/expo/eas-cli/blob/f21aa7d99bf14325d7c3ce1a18eb9289d4b5dece/node_modules/@types/prompts/index.d.ts#L107-L108